extended/src/main/java/io/kubernetes/client/workqueue/DefaultDelayingQueue.java

```diff
@@ -0,0 +1,70 @@
+package io.kubernetes.client.workqueue;
+
+import com.google.common.primitives.Longs;
+
+import java.time.Duration;
+import java.util.concurrent.DelayQueue;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The default delaying queue implementation.
```

Comment:
Please add more documentation about what this is and how it is used. We can't assume people are familiar with the client go approach.